samples/benchmarks/coremark/prj.conf

```diff
@@ -3,6 +3,8 @@
 #
 # SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
 
+CONFIG_NCS_SAMPLES_DEFAULTS=y
```

Comment:
This also implies `CONFIG_ASSERT` and HW stack protection. Maybe we could simply explicitly enable minimal logging to avoid enabling unwanted features accidentally?

Reply:
Yes, it does. Why are those features unwanted?

Shouldn't we strive to be aligned with other samples?

Reply:
While testing performance, generally assertions should not be enabled (assertions may affect performance measurements). HW stack protection may be enabled (should not impact performance).
Still, if a feature will be silently added to `CONFIG_NCS_SAMPLES_DEFAULTS` at some point, it may negatively affect our performance results, that's why I suggested to avoid using NCS samples defaults here

Reply:
ASSERTs do not affect performance as the CoreMark benchmark does not use them. On the other hand, asserts give us some additional safety measures before we start the benchmark and after we finish executing it.

Reply:
Zephyr may use asserts under the hood (e.g. when switching threads). Since we measure performance here, I would suggest to disable assertions because of that.